src/components/livestore/CustomLiveStoreProvider.tsx

```diff
@@ -77,13 +77,17 @@ export const CustomLiveStoreProvider: React.FC<
     syncPayload.current.authToken = accessToken || "";
   }, [clientId, accessToken]);
 
-  const adapter = makePersistedAdapter({
-    storage: { type: "opfs" },
-    worker: LiveStoreWorker,
-    sharedWorker: LiveStoreSharedWorker,
-    resetPersistence,
-    clientId, // This ties the LiveStore client to the authenticated user
-  });
+  const adapter = useMemo(
+    () =>
+      makePersistedAdapter({
+        storage: { type: "opfs" },
+        worker: LiveStoreWorker,
+        sharedWorker: LiveStoreSharedWorker,
+        resetPersistence,
+        clientId, // This ties the LiveStore client to the authenticated user
+      }),
+    [clientId, resetPersistence]
+  );
```

Comment:
Looking at https://github.com/livestorejs/livestore/blob/main/examples/src/web-todomvc/src/Root.tsx it seems like we should be setting this outside of here entirely. What I don't get though is if we need to pass the `clientId` on through here or if it can come in the sync payload _only_.